lib/collection/proxy-config.js

```diff
@@ -97,13 +124,37 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
      * @param {String=} [urlStr] The url string for which the proxy match needs to be done.
      */
     test: function(urlStr) {
-        // this ensures we don't proceed any further for any protocol
-        // that is not http or https
-        if (_.isEmpty(urlStr) || (_.isString(urlStr) && !urlStr.startsWith('http'))) {
+        // this ensures we don't proceed any further for any non-supported protocol
+        if (_.isEmpty(urlStr) || (_.isString(urlStr) && !urlStr.match(regexes.validProtocolTester))) {
             return false;
         }
 
         return this.match.test(urlStr);
+    },
+
+    /**
+     * Returns the proxy server url. The protocol in the returned URL will be taken from the passed `urlStr`.
+     * If the passed `urlStr` does not have any protocol then the default Protocol `http` will be used
+     * @param {String=} [urlStr] The url string for which the proxy will be used
+     * @returns {String}
+     */
+    getProxyUrl: function (urlStr) {
```

Comment:
It should gets the url as both postmanURL or string, since we allow that in proxyConfigList